### PR TITLE
rockchip: add kernel 5.10 support 

### PR DESCRIPTION
**Pull request by 1715173329**
This adds kernel 5.10 support for rockchip.

Link: https://patchwork.ozlabs.org/project/openwrt/patch/20210216224908.721491-1-mail@david-bauer.net/

Compile-tested on: rk3328 nanopi-r2s
Runtime-testd on: N/A

Signed-off-by: CN_SZTL <cnsztl@project-openwrt.eu.org>

### CODE REVIEW

File: target/linux/rockchip/armv8/config-5.4 [hunk 1]
```diff
@@ -115,72 +144,115 @@ CONFIG_CRC16=y
...
 CONFIG_CRYPTO_RNG_DEFAULT=y
+CONFIG_DCACHE_WORD_ACCESS=y
 CONFIG_DEBUG_BUGVERBOSE=y
 # CONFIG_DEVFREQ_GOV_PASSIVE is not set
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
 CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
-CONFIG_DEVFREQ_THERMAL=y
+# CONFIG_DEVFREQ_THERMAL is not set
```
Comment (AmadeusGhost) [inline target/linux/rockchip/armv8/config-5.4:173]: Why cancel this?

File: target/linux/rockchip/config-default [hunk 2]
```diff
@@ -1,297 +0,0 @@
-CONFIG_64BIT=y
```
Comment (AmadeusGhost) [inline target/linux/rockchip/config-default:1]: I don’t think it’s a good idea to move config-default to config-5.x, we should try to be consistent with upstream as much as possible.

File: target/linux/rockchip/files-5.10/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
```
Comment (AmadeusGhost) [inline target/linux/rockchip/files-5.10/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts:1]: This target still uses patch to add devices' dts, please keep consistent.

File: target/linux/rockchip/patches-5.10/200-rockchip-rk3399-Add-support-for-FriendlyARM-NanoPi-R.patch [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
++&fusb0 {
++	status = "disabled";
++};
++
++&pcie0 {
++	max-link-speed = <1>;
++	num-lanes = <1>;
++	vpcie3v3-supply = <&vcc3v3_sys>;
++
++	pcie@0 {
```
Comment (AmadeusGhost) [inline target/linux/rockchip/patches-5.10/200-rockchip-rk3399-Add-support-for-FriendlyARM-NanoPi-R.patch:159]: Traditionally, it’s better to change this to "pcie@0,0".

File: target/linux/rockchip/files-5.10/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts (hunk 3 above)
Comment (1715173329) [reply target/linux/rockchip/files-5.10/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts:1]: This is a hack made by upstream, in my option it's fine to keep it as-is.

File: target/linux/rockchip/config-default (hunk 2 above)
Comment (1715173329) [reply target/linux/rockchip/config-default:1]: First, it's actually made by upstream.
Second, there're some differences between 5.4 and 5.10, so it's the right approach.

File: target/linux/rockchip/armv8/config-5.4 (hunk 1 above)
Comment (1715173329) [reply target/linux/rockchip/armv8/config-5.4:173]: This is useless, at least for now.

File: target/linux/rockchip/patches-5.10/200-rockchip-rk3399-Add-support-for-FriendlyARM-NanoPi-R.patch (hunk 4 above)
Comment (1715173329) [reply target/linux/rockchip/patches-5.10/200-rockchip-rk3399-Add-support-for-FriendlyARM-NanoPi-R.patch:159]: This is not right, see:
https://github.com/armbian/build/blob/master/patch/kernel/rockchip64-dev/add-board-nanopi-r4s.patch#L114

**Comment by 1715173329** [comment]
Ping @AmadeusGhost.

**Comment by 1715173329** [comment]
Merged.

**Pull request closed without merging**